workflow/controller/operator.go

```diff
@@ -1117,6 +1124,18 @@ func (woc *wfOperationCtx) podReconciliation(ctx context.Context) error {
 		node, err := woc.wf.Status.Nodes.Get(nodeID)
 		if err == nil {
 			if newState := woc.assessNodeStatus(pod, node); newState != nil {
+				if newState.Succeeded() {
+					tmpl, err := woc.GetNodeTemplate(node)
+					if err != nil {
+						return
+					}
+					// Check whether the node has output and whether its taskresult is in an incompleted state.
+					if tmpl.HasOutputs() && woc.wf.Status.IsTaskResultInCompleted(node.ID) && woc.wf.Status.IsTaskResultInCompleted(pod.Name) {
+						woc.log.WithFields(log.Fields{"nodeID": newState.ID}).WithError(err).Error("Taskresult of the node not yet completed")
```

Comment:
Same comment along the lines of what @juliev0 was saying, I don't think this is an error. We just need to flag `needReconcileTaskResult`. Could maybe just log it normally if you want the log?

Reply:
right, probably a Debug line

Reply:
By the way, do we need to call it for both `woc.wf.Status.IsTaskResultInCompleted(node.ID) && woc.wf.Status.IsTaskResultInCompleted(pod.Name)`?

Reply:
Oh yeah, I was thinking this but had to step away and forgot to ask. I think it should just be `tmpl.HasOutputs() && woc.wf.Status.IsTaskResultInCompleted(node.ID)`

Reply:
Maybe confusion from the comment [here](https://github.com/argoproj/argo-workflows/pull/12537#discussion_r1464003132)

Reply:
seeing that comment about the comment made me think of it :)

Reply:
> By the way, do we need to call it for both `woc.wf.Status.IsTaskResultInCompleted(node.ID) && woc.wf.Status.IsTaskResultInCompleted(pod.Name)`?

Yes, I thought about this problem at first. But there is a problem. If the outputs are in pod annotations or in taskresult, the key values ​​are different. Maybe we can unify to podName or NodeId. I think nodeId is better, how about you?

May be I can add a func named pod.GetNodeId()

```
	if x, ok := pod.Annotations[common.AnnotationKeyReportOutputsCompleted]; ok {
		woc.log.Warn("workflow uses legacy/insecure pod patch, see https://argo-workflows.readthedocs.io/en/latest/workflow-rbac/")
		resultName := pod.GetName()
		if x == "true" {
			woc.wf.Status.MarkTaskResultComplete(resultName)
		} else {
			woc.wf.Status.MarkTaskResultIncomplete(resultName)
		}
	}

Reply:
Yes, just unify to Node ID.